### PR TITLE
Replace package satori/go.uuid for cloudprovider ionoscloud

### LINKED ISSUE #4068: Addressing CVE-2021-3538 with satori/go.uuid
There's an unresolved CVE that impacts the satori/go.uuid library. I noticed the two providers using the library.

Should probably switch to a maintained UUID library like https://github.com/google/uuid or something.

Tagging owners @ringtail @avorima @schegi

### PR DESCRIPTION
**Pull request by avorima**
Fixes #4068  for ionoscloud provider.

### CODE REVIEW

**Review by feiskyer — COMMENTED** [review]
LGTM

/approve

File: cluster-autoscaler/go.mod [hunk 1]
```diff
@@ -10,11 +10,10 @@ require (
 	github.com/Azure/go-autorest/autorest/date v0.3.0
 	github.com/Azure/go-autorest/autorest/to v0.2.0
 	github.com/aws/aws-sdk-go v1.35.24
-	github.com/codegangsta/negroni v1.0.0 // indirect
 	github.com/digitalocean/godo v1.27.0
 	github.com/ghodss/yaml v1.0.0
 	github.com/golang/mock v1.4.4
-	github.com/gorilla/context v1.1.1 // indirect
+	github.com/google/uuid v1.2.0
```
Comment (mwielgus) [inline cluster-autoscaler/go.mod:16]: Is it possible to use 1.1.2 ? Kubernetes/kubernetes uses 1.1.2 and we would rather like to stay consistent.

**Review by mwielgus — CHANGES_REQUESTED** [review]

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (avorima) [reply cluster-autoscaler/go.mod]: Done

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**